test/scripts/kind_dual_conversion.sh

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env bash
+set -x
```

Comment:
this is not meant to do the testing of the feature. is just a helper for the developer.
I will move it to contrib/ to avoid misunderstanding.
For the testing, this needs more work and we should avoid to do it in bash ... these things always end in a 10k script that nobody wants to touch